x/ibc/06-solomachine/types/consensus_state.go

```diff
@@ -0,0 +1,52 @@
+package types
+
+import (
+	"github.com/tendermint/tendermint/crypto"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	clientexported "github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
+	clienttypes "github.com/cosmos/cosmos-sdk/x/ibc/02-client/types"
+	commitmentexported "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment/exported"
+)
+
+var _ clientexported.ConsensusState = ConsensusState{}
+
+// ConsensusState defines a Solo Machine consensus state
+type ConsensusState struct {
+	Sequence uint64 `json:"sequence" yaml:"sequence"`
+
+	PubKey crypto.PubKey `json:"pubkey" yaml: "pubkey"`
+}
+
+// ClientType returns Solo Machine type
+func (ConsensusState) ClientType() clientexported.ClientType {
+	return clientexported.SoloMachine
+}
+
+// GetHeight returns the sequence number
+func (cs ConsensusState) GetHeight() uint64 {
+	return cs.Sequence
+}
+
+// GetTimestamp returns zero.
+func (cs ConsensusState) GetTimestamp() uint64 {
```

Comment:
should solo machines support timestamps?

Reply:
The signed message should include a timestamp, for timestamp-based timeouts to work